whois-api/src/main/java/net/ripe/db/whois/api/rest/WhoisSearchService.java

```diff
@@ -214,7 +218,9 @@ private void validateSearchKey(final HttpServletRequest request, final String se
 
     private void validateSources(final HttpServletRequest request, final Set<String> sources) {
         for (final String source : sources) {
-            if (!sourceContext.getAllSourceNames().contains(ciString(source))) {
+            // NONAUTH source object is in main source (RIPE source)
```

Comment:
Can you move this logic to SourceContext? Something like isOutOfRegion(String source) and isNonAuthSource(String source)? I guess we'll need to use it from multiple places

Reply:
Thank you. I knew it could be simplified, but I was not able to see it myself :)